x-pack/plugins/security_solution/public/common/components/search_bar/index.tsx

```diff
@@ -479,6 +471,41 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
     dispatch(inputsActions.setSearchBarFilter({ id, filters })),
 });
 
+const useSyncSearchBarUrlParam = () => {
+  const updateSavedQueryUrlParam = useUpdateUrlParam<string>(CONSTANTS.savedQuery);
+  const updateAppQueryUrlParam = useUpdateUrlParam<Query>(CONSTANTS.appQuery);
+  const updateFilterUrlParam = useUpdateUrlParam<Filter[]>(CONSTANTS.filters);
+
+  const getGlobalQuerySelector = useMemo(() => inputsSelectors.globalQuerySelector(), []);
+  const getGlobalFiltersQuerySelector = useMemo(
+    () => inputsSelectors.globalFiltersQuerySelector(),
+    []
+  );
+  const getGlobalSavedQuerySelector = useMemo(() => inputsSelectors.globalSavedQuerySelector(), []);
+
+  const query = useSelector(getGlobalQuerySelector);
+  const filters = useSelector(getGlobalFiltersQuerySelector);
+  const savedQuery = useSelector(getGlobalSavedQuerySelector);
+
+  useEffect(() => {
+    if (savedQuery != null && savedQuery.id !== '') {
```

Comment:
Add this code to reproduce previous behavior. Only the saved query gets to the URL when there is a saved query.